### DISCUSSION TITLE
questions regarding analyzing heart period from ECG data

### DISCUSSION THREAD

**Discussion by yingjieQ** [bachlab/PsPM#583] [Q&A] [answered]
Hi,

I'm trying to use a one-channel ECG data which is recorded in a .ncs format to perform within-subject statistical analyses on heart period for a fear conditioning paradigm. 

Clearly, this format is not among those data type that can be read by built-in pspm scripts, I tried to check pspm manual but didn't really find what general formats are supposed to feed in to import/preprocess the ECG. 

What I can do and already did is to transform the ECG.ncs into .mat formats, and in there I have the raw sampled data with both positive and negative values. I also have separate event files enabling cutting out ECG segments for each trial. The idea is to separate ECG of different trial types from the same subject, and compare the heart period from different trial types. 

Also a related problem, I downloaded and installed PsPM 6.1 on a linux system last December, but when I tried to open pspm_ecg2hb.m, it's not in the directory unzipped from the package. The only ecg related .m files being found is ecg_editor. 

Thank you in advance for your help!

Best,
Yingjie

**Comment by dominikbach** [comment, marked as answer]
Hi Yingjie

thanks for posting this. Indeed, *.ncs format is not currently supported by PsPM.

If there is an open-source matlab converter we might be able to write an import routine but this is likely to take a while.

The PsPM manual describes how to important matlab or text data on page 42 of the latest release (6.1.1).

This also fixes the problem of not finding the ECG2HB routine. (The function is in fact called pspm_convert_ecg2hb.m).

Hope this helps
Dominik
